src/communication/RpcServer.cpp

```diff
@@ -10,3 +10,116 @@
+	// Validate the method name using a URI validator.
+	auto [valid, reason] = Validator::uri::isValidRpcMethod(method_name);
+	if (!valid) {
+		// If the method name is invalid, return an error status.
+		v1::UStatus status;
+		status.set_code(v1::UCode::INVALID_ARGUMENT);
+		status.set_message("Invalid rpc URI");
+		return uprotocol::utils::Unexpected(status);
+	}
+
+	// Validate the payload format, if provided.
+	if (payload_format.has_value()) {
+		if (!UPayloadFormat_IsValid(payload_format.value())) {
+			// If the payload format is invalid, return an error status.
+			v1::UStatus status;
+			status.set_code(v1::UCode::OUT_OF_RANGE);
+			status.set_message("Invalid payload format");
+			return uprotocol::utils::Unexpected(status);
+		}
+	}
+
+	// Create the RpcServer instance with the provided parameters.
+	auto server = std::make_unique<RpcServer>(
+	    std::forward<std::shared_ptr<transport::UTransport>>(transport),
+	    std::forward<std::optional<v1::UPayloadFormat>>(payload_format),
+	    std::forward<std::optional<std::chrono::milliseconds>>(ttl));
+
+	// Attempt to connect the server with the provided method name and callback.
+	auto status = server->connect(method_name, std::move(callback));
+	if (status.code() == v1::UCode::OK) {
+		// If connection is successful, return the server instance.
+		return server;
+	} else {
+		// If connection fails, return the error status.
+		return uprotocol::utils::Unexpected(std::move(status));
+	}
+}
+
+v1::UStatus RpcServer::connect(const v1::UUri& method, RpcCallback&& callback) {
+	callback_ = std::move(callback);
+	auto result = transport_->registerListener(
+	    // sink_filter=
+	    method,
+	    // listener=
+	    [this](const v1::UMessage& request) {
+		    // Validate the request message using a RPC message validator.
+		    auto [valid, reason] =
+		        Validator::message::isValidRpcRequest(request);
+		    if (!valid) {
+			    return;
+		    }
+
+		    // Create a response message builder using the request message.
+		    auto builder =
+		        datamodel::builder::UMessageBuilder::response(request);
+
+		    // Call the RPC callback method with the request message.
+		    auto payloadData = callback_(request);
+
+		    if (ttl_.has_value()) {
+			    builder.withTtl(ttl_.value());
+		    }
+
+		    if (expected_payload_format_.has_value()) {
+			    builder.withPayloadFormat(expected_payload_format_.value());
+		    }
+
+		    // Check for payload data requirement based on expected format
+		    // presence
+		    if (!payloadData.has_value()) {
+			    // builder.build() verifies if payload format is required
+			    auto response = builder.build();
+			    // Ignoring status code for transport send
+			    std::ignore = transport_->send(response);
+		    } else {
+			    // builder.build(payloadData) verifies if payload format
+			    // matches the expected
+			    auto response = builder.build(std::move(payloadData.value()));
```

Comment:
Minor: I _think_ this needs to be
```suggestion
			    auto response = builder.build(std::move(payloadData).value());
```
in order for the move to actually work.

Reply:
Done